langchain/src/stores/message/planetscale.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Client as PlanetscaleClient,
+  Config as PlanetscaleConfig,
+  Connection as PlanetscaleConnection,
+} from "@planetscale/database";
+import {
+  BaseMessage,
+  BaseListChatMessageHistory,
+  StoredMessage,
+  StoredMessageData,
+} from "../../schema/index.js";
+import {
+  mapChatMessagesToStoredMessages,
+  mapStoredMessagesToChatMessages,
+} from "./utils.js";
+
+export type PlanetscaleChatMessageHistoryInput = {
+  tableName?: string;
+  sessionId: string;
+  config?: PlanetscaleConfig;
+  client?: PlanetscaleClient;
+};
+
+interface selectStoredMessagesDTO {
+  id: string;
+  session_id: string;
+  type: string;
+  content: string;
+  role: string | null;
+  name: string | null;
+  additional_kwargs: string;
+}
+
+export class PlanetscaleChatMessageHistory extends BaseListChatMessageHistory {
+  lc_namespace = ["langchain", "stores", "message", "planetscale"];
+
+  get lc_secrets() {
+    return {
+      "config.host": "PLANETSCALE_HOST",
+      "config.username": "PLANETSCALE_USERNAME",
+      "config.password": "PLANETSCALE_PASSWORD",
+      "config.url": "PLANETSCALE_DATABASE_URL",
+    };
+  }
+
+  public client: PlanetscaleClient;
+
+  private connection: PlanetscaleConnection;
+
+  private tableName: string;
+
+  private sessionId: string;
+
+  constructor(fields: PlanetscaleChatMessageHistoryInput) {
+    super(fields);
+
+    const { sessionId, config, client, tableName } = fields;
+
+    if (client) {
+      this.client = client;
+    } else if (config) {
+      this.client = new PlanetscaleClient(config);
+    } else {
+      throw new Error(
+        "Either a client or config must be provided to PlanetscaleChatMessageHistory"
+      );
+    }
+
+    this.connection = this.client.connection();
+
+    this.tableName = tableName || "stored_message";
+    this.sessionId = sessionId;
+  }
+
+  private async createTableIfNotExists(): Promise<void> {
+    const query = `CREATE TABLE IF NOT EXISTS ${this.tableName} (id BINARY(16) PRIMARY KEY, session_id VARCHAR(255), type VARCHAR(255), content VARCHAR(255), role VARCHAR(255), name VARCHAR(255), additional_kwargs VARCHAR(255));`;
```

Comment:
nit: Store `additional_kwargs` as a JSON field for potential easier querying down the road?